staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go

```diff
@@ -51,3 +48,14 @@ func (oa OpenAPI) Install(c *restful.Container, mux *mux.PathRecorderMux) (*hand
 
 	return openAPIVersionedService, spec
 }
+
+// BuildAndPruneOpenAPISpec builds an OpenAPI spec from
+// the give RegisteredWebServices and OpenAPI config
+func BuildAndPruneOpenAPISpec(s []*restful.WebService, cfg *common.Config) *spec.Swagger {
+	spec, err := builder.BuildOpenAPISpec(s, cfg)
+	if err != nil {
+		klog.Fatalf("Failed to build open api spec for root: %v", err)
```

Comment:
change the signature to `*spec.Swagger, error` and `return nil, err` here

can continue to klog.Fatal in Install, but this gives callers the choice about how to handle the error

Reply:
done. I'm also now calling klog.Fatal in `PrepareRun` in order to mimic the previous behavior (when the spec was only built bay calling Install), but let me know if a klog.Error is more appropriate here.